pkg/remote/bound/transmeta_bound_test.go

```diff
@@ -0,0 +1,211 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bound
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/cloudwego/kitex/internal/test"
+	"github.com/cloudwego/kitex/pkg/consts"
+	"github.com/cloudwego/kitex/pkg/remote"
+	"github.com/cloudwego/kitex/pkg/rpcinfo"
+	"github.com/cloudwego/kitex/pkg/rpcinfo/remoteinfo"
+
+	remoteMocks "github.com/cloudwego/kitex/internal/mocks/remote"
+	"github.com/cloudwego/kitex/pkg/remote/trans/invoke"
+)
+
+// TestNewTransMetaHandler test NewTransMetaHandler function and assert the result not nil.
+func TestNewTransMetaHandler(t *testing.T) {
+	metaHandler1 := &remoteMocks.MetaHandler{}
+	metaHandler2 := &remoteMocks.MetaHandler{}
+	metaHandler3 := &remoteMocks.MetaHandler{}
+	mhs := []remote.MetaHandler{
+		metaHandler1, metaHandler2, metaHandler3,
+	}
+
+	handler := NewTransMetaHandler(mhs)
+
+	test.Assert(t, handler != nil)
+}
+
+// TestTransMetaHandlerWrite test Write function of transMetaHandler.
+func TestTransMetaHandlerWrite(t *testing.T) {
+	t.Run("Test transMetahandler Write success", func(t *testing.T) {
+		metaHandler1 := &remoteMocks.MetaHandler{}
+		metaHandler2 := &remoteMocks.MetaHandler{}
+		metaHandler3 := &remoteMocks.MetaHandler{}
+		mhs := []remote.MetaHandler{
+			metaHandler1, metaHandler2, metaHandler3,
+		}
+
+		invokeMessage := invoke.NewMessage(nil, nil)
+		remoteMessage := remote.NewMessage(
+			nil, nil, nil, remote.Call, remote.Client)
+		ctx := context.Background()
+
+		handler := NewTransMetaHandler(mhs)
+
+		metaHandler1.On("WriteMeta", ctx, remoteMessage).Return(context.Background(), nil)
+		metaHandler2.On("WriteMeta", ctx, remoteMessage).Return(context.Background(), nil)
+		metaHandler3.On("WriteMeta", ctx, remoteMessage).Return(context.Background(), nil)
+
+		test.Assert(t, handler != nil)
+		ctx, err := handler.Write(ctx, invokeMessage, remoteMessage)
+		test.Assert(t, ctx != nil)
+		test.Assert(t, err == nil)
+	})
+
+	t.Run("Test TransMetahandler Write with error", func(t *testing.T) {
+		metaHandler1 := &remoteMocks.MetaHandler{}
+		metaHandler2 := &remoteMocks.MetaHandler{}
+		metaHandler3 := &remoteMocks.MetaHandler{}
+		mhs := []remote.MetaHandler{
+			metaHandler1, metaHandler2, metaHandler3,
+		}
+
+		invokeMessage := invoke.NewMessage(nil, nil)
+		remoteMessage := remote.NewMessage(
+			nil, nil, nil, remote.Call, remote.Client)
+		ctx := context.Background()
+
+		metaHandler1.On("WriteMeta", ctx, remoteMessage).Return(context.Background(), errFoo)
+		metaHandler2.On("WriteMeta", ctx, remoteMessage).Return(context.Background(), nil)
+		metaHandler3.On("WriteMeta", ctx, remoteMessage).Return(context.Background(), nil)
+
+		handler := NewTransMetaHandler(mhs)
+		test.Assert(t, handler != nil)
+
+		ctx, err := handler.Write(context.Background(), invokeMessage, remoteMessage)
+		test.Assert(t, ctx != nil)
+		test.Assert(t, errors.Is(errFoo, err))
+	})
```

Comment:
could u pls add a test case that set k-v to ctx then check it after Write is called, maybe like this
```
metaHandler1.On("WriteMeta", mock2.MatchedBy(func(context.Context) bool { return true }), mock2.MatchedBy(func(remote.Message) bool { return true })).Return(func(ctx context.Context, msg remote.Message) context.Context {
		ctx = context.WithValue(ctx, "key1", "val1")
		return ctx
	}, nil)
metaHandler2.On("WriteMeta", mock2.MatchedBy(func(context.Context) bool { return true }), mock2.MatchedBy(func(remote.Message) bool { return true })).Return(func(ctx context.Context, msg remote.Message) context.Context {
		ctx = context.WithValue(ctx, "key2", "val2")
		return ctx
	}, nil)
...
newctx, err := handler.Write(ctx, mocks.Conn{}, remoteMessage)
test.Assert(t, err == nil, err)
mock := newctx.Value("key1")
test.Assert(t, mock == "val1")
mock1 := newctx.Value("key2")
test.Assert(t, mock1 == "val2")

```

Reply:
and add same case for OnMessage ; )